### ISSUE TITLE
Incorrect error count when using the unsupported unary plus

### ISSUE THREAD

**Issue by 0xmohit** [rust-lang/rust#36499]
Executing the following:

``` rust
fn main() {
  2 + +2;
}
```

results in:

```
foo.rs:2:7: 2:8 error: expected expression, found `+`
foo.rs:2   2 + +2;
               ^
error: aborting due to 2 previous errors
```

The [reference](https://doc.rust-lang.org/reference.html#unary-operator-expressions) makes it clear that the above code would be rejected.  However, the error count indicated doesn't appear to be correct.

**Comment by TimNN** [comment]
Introduced between `nightly-2016-04-22` and `nightly-2016-04-25` ([Changes](http://github.com/rust-lang/rust/compare/b5ba5923f...645dd013a))

I'm pretty sure I found the problem and should have a fix ready soon.
